spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinAutoConfiguration.java

```diff
@@ -31,25 +28,21 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import com.github.kristofa.brave.SpanCollector;
 
 /**
  * @author Spencer Gibb
  */
 @Configuration
 @EnableConfigurationProperties
-@ConditionalOnClass(SpanCollector.class)
 @ConditionalOnProperty(value = "spring.zipkin.enabled", matchIfMissing = true)
 public class ZipkinAutoConfiguration {
 
 	@Bean
-	@ConditionalOnMissingBean(SpanCollector.class)
-	public SpanCollector spanCollector() {
+	@ConditionalOnMissingBean(ZipkinSpanReporter.class)
+	public ZipkinSpanReporter reporter() {
 		ZipkinProperties zipkin = zipkinProperties();
 		String url = "http://" + zipkin.getHost() + ":" + zipkin.getPort();
```

Comment:
yes it is common, and the point of this change wasn't to change to https it was to replace thrift spans with json ones. We can revisit how urls are specified of course, but I think we are getting distracted.